File: app/src/main/java/com/zxcx/shitang/ui/search/result/SearchResultPresenter.java
package com.zxcx.shitang.ui.search.result;

import android.support.annotation.NonNull;

import com.zxcx.shitang.mvpBase.BasePresenter;

import java.util.List;

public class SearchResultPresenter extends BasePresenter<SearchResultContract.View> implements SearchResultContract.Presenter {

    private final SearchResultModel mModel;

    public SearchResultPresenter(@NonNull SearchResultContract.View view) {
        attachView(view);
        mModel = new SearchResultModel(this);
    }

    public void searchCard(String keyword, int page, int pageSize){
        mModel.searchCard(keyword,page, pageSize);
    }

    public void searchCardBag(String keyword){
        mModel.searchCardBag(keyword);
    }

    @Override
    public void getDataSuccess(List<SearchCardBean> bean) {
        mView.getDataSuccess(bean);
    }

    @Override
    public void searchCardBagSuccess(List<SearchCardBagBean> list) {
        mView.searchCardBagSuccess(list);
    }

    @Override
    public void getDataFail(String msg) {
        mView.toastFail(msg);
    }

    @Override
    public void showLoading() {
        mView.showLoading();
    }

    @Override
    public void hideLoading() {
        mView.hideLoading();
    }

    @Override
    public void startLogin() {
        mView.startLogin();
    }

    public void detachView() {
        super.detachView();
        mModel.onDestroy();
    }
}

File: app/src/main/java/com/zxcx/zhizhe/ui/search/result/SearchResultPresenter.kt
package com.zxcx.zhizhe.ui.search.result

import com.zxcx.zhizhe.mvpBase.BasePresenter

class SearchResultPresenter(view: SearchResultContract.View) : BasePresenter<SearchResultContract.View>(), SearchResultContract.Presenter {

    private val mModel: SearchResultModel

    init {
        attachView(view)
        mModel = SearchResultModel(this)
    }

    fun search(keyword: String, page: Int, pageSize: Int) {
        mModel.search(keyword, page, pageSize)
    }

    override fun getDataSuccess(bean: MutableList<SearchResultBean>) {
        mView.getDataSuccess(bean)
    }

    override fun getDataFail(msg: String) {
        mView.toastFail(msg)
    }

    override fun showLoading() {
        mView.showLoading()
    }

    override fun hideLoading() {
        mView.hideLoading()
    }

    override fun startLogin() {
        mView.startLogin()
    }

    override fun detachView() {
        super.detachView()
        mModel.onDestroy()
    }
}

